Tests/GraphitiTests/ValidationRulesTests.swift

```diff
@@ -0,0 +1,68 @@
+import Foundation
+@testable import Graphiti
+import GraphQL
+import NIO
+import XCTest
+
+class ValidationRulesTests: XCTestCase {
+    // Tests that circularly dependent objects can be used in schema and resolved correctly
+    func testRegisteringCustomValidationRule() throws {
+        struct TestResolver {
+            var helloWorld: String { "Hellow World" }
+        }
+        
+        let testSchema = try Schema<TestResolver, NoContext> {
+            Query {
+                Field("helloWorld", at: \.helloWorld)
+            }
+        }
+        let api = TestAPI<TestResolver, NoContext>(
+            resolver: TestResolver(),
+            schema: testSchema,
+            validationRules: GraphQL.specifiedRules + [disableIntrospectionValidation(context:)]
```

Comment:
I was debating myself if that makes sense. I'm not sure. 
What if you don't want to have any rules applied in the validation step? That would need another API to do so. We could work with a `ValidationConfiguration` struct that contains a property `applyDefaultRules` that is `true` by default and can be changed to `false`.
What's your opinion on this?

Reply:
Cool, thanks for walking me through your thoughts. I guess I'm in favor of always applying the `specifiedRules` validations, since they are intended to match [the validations called out in the GraphQL spec](https://spec.graphql.org/October2021/#sec-Validation). While there definitely could be times that a user would want to break spec compliance, I think it'd be pretty uncommon.

Reply:
Alright, thanks for your opinion.